internal/input/changes.go

```diff
@@ -116,97 +110,205 @@ func closeChannels(accChannel, offChannel, trustChannel chan xdr.LedgerEntry) {
 	}
 }
 
-// sendLedgerChangesToChannels streams the changes from a single ledger to the provided channels
-func sendLedgerChangesToChannels(seqNum uint32, core *ledgerbackend.CaptiveStellarCore, accChannel, offChannel, trustChannel chan xdr.LedgerEntry) {
-	changeReader, err := ingestio.NewLedgerChangeReader(core, network.PublicNetworkPassphrase, seqNum)
-	if err == nil {
-		compactedChanges := getCompactedChanges(changeReader)
-		for _, change := range compactedChanges {
-			// TODO: figure out how to handle deleted entries
-			if change.Post != nil {
-				entry := *change.Post
-				sendToChannel(entry, accChannel, offChannel, trustChannel)
+func addLedgerChangesToCache(changeReader *ingestio.LedgerChangeReader, accCache, offCache, trustCache *ingestio.LedgerEntryChangeCache) error {
+	for {
+		change, err := changeReader.Read()
+		if err == ingestio.EOF {
+			break
+		}
+
+		if err != nil {
+			return err
+		}
+
+		switch change.Type {
+		case xdr.LedgerEntryTypeAccount:
+			accCache.AddChange(change)
+
+		case xdr.LedgerEntryTypeOffer:
+			offCache.AddChange(change)
+
+		case xdr.LedgerEntryTypeTrustline:
+			trustCache.AddChange(change)
+
+		default:
```

Comment:
Is there a `default` case not covered by the above cases? If so and it's a rare event, might be worth adding some logging here. If so and it's not rare, add a comment explaining this case.